omg/core/views.py

```diff
@@ -120,30 +136,44 @@ def element_icons(self):
         else:
             return None
 
-    def _load_element_icons(self):
-        for element_type in self.player.elements:
-            icon_texture = arcade.load_texture(element_type["image_file"])
-            self.element_icons.append(icon_texture)
-
     def on_draw(self):
         """Drawing code."""
-        arcade.start_render()
+        self.clear()
+
+        # Activate player camera to draw Sprites
+        # sprites outside of the player camera are not drawn
+        self.camera_sprite.use()
         self.player.draw()
-        self.obstacles.draw()
-        self.projectiles.draw()
-        self.pickupables.draw()
+        self.scene.draw()
+
+        if self.debug_mode:
+            self.scene.draw_hit_boxes(arcade.color.RED)
+            self.player.draw_hit_box(arcade.color.RED)
+        # Activate GUI camera before drawing GUI elements
+        # This is to ensure GUI elements are drawn w.r.t the window
+        self.camera_gui.use()
         self._draw_ui()
+        if self.debug_mode:
+            self._draw_debug_info()
 
     def update(self, delta_time):
         """Main update window."""
-        self.player.update(self.mouse_x, self.mouse_y, delta_time)
-        self.obstacles.update()
+        # Mouse is tracked in the window coordinates, but the player logic needs
+        # the mouse in the camera coordinates. For example, if the player moves
+        self.scene.update()
+        # to the right but mouse is kept constant, that is a mouse movement to
+        # the right as far as the player is concerned.
```

Comment:
This comment is kinda confusing, if possible can you make it more clear?

Reply:
done